tests/CreatesApplication.php

```diff
@@ -2,21 +2,52 @@
 
 namespace Tests;
 
+use DoSomething\Gateway\Northstar;
 use Illuminate\Contracts\Console\Kernel;
+use DoSomething\Gateway\Resources\NorthstarUser;
 
 trait CreatesApplication
 {
+    /**
+     * The Northstar API client mock.
+     *
+     * @var \Mockery\MockInterface
+     */
+    protected $northstarMock;
+
+    /**
+     * The Faker generator, for creating test data.
+     *
+     * @var \Faker\Generator
+     */
+    protected $faker;
+
     /**
      * Creates the application.
      *
      * @return \Illuminate\Foundation\Application
      */
     public function createApplication()
     {
-        $app = require __DIR__.'/../bootstrap/app.php';
+        $this->app = require __DIR__.'/../bootstrap/app.php';
+
+        $this->app->make(Kernel::class)->bootstrap();
 
-        $app->make(Kernel::class)->bootstrap();
+        $this->northstarMock = $this->mock(Northstar::class);
+        $this->northstarMock->shouldReceive('asClient')->andReturnSelf();
```

Comment:
@DFurnes I'm still getting a lot of these errors for Post Tests and Signup Tests:
```
"message" => "Method Mockery_0_DoSomething_Gateway_Northstar::asClient() does not exist on this mock object"
```
however, it does exist here. Do I need to call this in a special way or anything from the `PostTest.php` / `SignupTest.php` files?

Reply:
Since we're calling `$this->mock(Northstar::class)` in each of these tests we're rewriting the mock created here. Instead, let's use the existing mock (we can remove all of these in `PostTest` and `SignupTest` and use the base mock we've created here).

If we ever want a one-off expectation applied to this mock, instead of `$this->mock(Northstar::class)`, we can just overwrite the existing mock at `$this->northstarMock`.

Reply:
OHH I get it now! Thank you @DFurnes ! All is updated :)